solidity/src/CadenceRandomConsumer.sol

```diff
@@ -69,27 +193,63 @@ abstract contract CadenceRandomConsumer is CadenceArchWrapper {
      * @param requestId The ID of the randomness request to fulfill.
      * @return randomResult The random value generated from the Flow block.
      */
-    function _fulfillRandomness(uint32 requestId) internal returns (uint64) {
+    function _fulfillRandomness(uint256 requestId) private returns (bytes32) {
         // Get the request details. Recall that request IDs are 1-indexed to allow for 0 to be used as an invalid value
-        uint32 requestIndex = requestId - 1;
+        uint256 requestIndex = requestId - 1;
         require(requestIndex < _requests.length, "Invalid request ID - value exceeds the number of existing requests");
-        Request memory request = _requests[requestIndex];
 
-        // Ensure that the request is fulfilled at a Flow block height greater than the one at which the request was made
-        uint64 flowHeight = _flowBlockHeight();
-        require(request.flowHeight < flowHeight, "Cannot fulfill request until subsequent Flow network block height");
+        // Access & validate the request
+        Request storage request = _requests[requestIndex];
+        _validateRequest(request);
+        request.fulfilled = true; // Mark the request as fulfilled
 
-        // Get the random source for the Flow block at which the request was made
-        uint64 randomResult = _getRandomSource(request.flowHeight); // returns bytes32
-        // Pack the randomResult into a uint64, hashing with the requestId to vary results across shared block heights.
-        // The random seed returned from Cadence Arch is only 32 bytes. Here only 8 bytes were required, but if an
-        // implementing contract requires more random bytes, the source should be expanded into any arbitrary number of
-        // bytes using a PRG.
-        randomResult = uint64(uint256(keccak256(abi.encodePacked(randomResult, requestId))));
+        // Get the random source for the Flow block at which the request was made, emit & return
+        bytes32 randomSource = _getRandomSource(request.flowHeight);
 
-        emit RandomnessFulfilled(requestId, request.flowHeight, request.evmHeight, randomResult);
+        emit RandomnessSourced(requestId, request.flowHeight, request.evmHeight, randomSource);
 
-        // Return the random result
-        return randomResult;
+        return randomSource;
+    }
+
+    /**
+     * @dev This method aggregates 256 bits of randomness by calling _revertibleRandom() 4 times.
+     *
+     * @return randomValue The aggregated 256 bits of randomness.
+     */
+    function _aggregateRevertibleRandom256() private view returns (uint256) {
+        // Call _revertibleRandom() 4 times to aggregate 256 bits of randomness
+        uint256 randomValue = uint256(_revertibleRandom());
```

Comment:
In the case of `revertibleRandom`, there is no need to create a PRG, because `revertibleRandom` already behaves like one (actually there is an internal PRG in FVM). Each time you call it, the PRG squeezes out a new random